http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -530,6 +532,53 @@ class RuntimeServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
                                            Some(traceId))
         publisherQueue.enqueue1(message).as(message)
       }
+      // we only need to return the message that might cause a start/stop transition
+    } yield msg
+
+  private[service] def processUpdateGceWithPdConfigRequest(req: UpdateRuntimeConfigRequest.GceConfig,
+                                                           allowStop: Boolean,
+                                                           runtime: Runtime,
+                                                           gceConfig: RuntimeConfig.GceWithPdConfig,
+                                                           traceId: TraceId): F[Option[UpdateRuntimeMessage]] =
+    for {
+      // should machine type be updated?
+      targetMachineType <- traverseIfChanged(req.updatedMachineType, gceConfig.machineType) { mt =>
+        if (runtime.status == RuntimeStatus.Stopped)
+          Async[F].pure((mt, false))
+        else if (!allowStop)
+          Async[F].raiseError[(MachineTypeName, Boolean)](RuntimeMachineTypeCannotBeChangedException(runtime))
+        else Async[F].pure((mt, true))
+      }
+      // should disk size be updated?
+      diskId <- F.fromEither(
+        gceConfig.persistentDiskId.toRight(RuntimeDiskNotFound(runtime.googleProject, runtime.runtimeName, traceId))
+      )
+      diskOpt <- persistentDiskQuery.getById(diskId).transaction
+      disk <- F.fromEither(diskOpt.toRight(DiskNotFoundByIdException(diskId, traceId)))
+      targetDiskSize <- traverseIfChanged(req.updatedDiskSize, disk.size) { d =>
```

Comment:
while you're in here, can you test to see if disk size change require restart? (there was a bug ticket that updating disk size didn't take effect until machine is restarted

Reply:
can you elaborate on the bug? I just tried updating a disk size and it looks like it went through fine, but was the machine unable to use the disk?

Reply:
https://broadworkbench.atlassian.net/browse/IA-2015 is the ticket...